src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.Linux.cs

```diff
@@ -101,25 +104,64 @@ private static void SetSslOptionsForSupportedBackend(EasyRequest easy, ClientCer
                 }
             }
 
+            private static void GetSslCaLocations(out string path, out string info)
+            {
+                // We only provide curl option values when SSL_CERT_FILE or SSL_CERT_DIR is set.
+                // When that is the case, we set the options so curl ends up using the same certificates as the
+                // X509 machine store.
+                path = _sslCaPath;
+                info = _sslCaInfo;
+
+                if (path == null || info == null)
+                {
+                    bool hasEnvironmentVariables = Environment.GetEnvironmentVariable("SSL_CERT_FILE") != null ||
+                                                   Environment.GetEnvironmentVariable("SSL_CERT_DIR") != null;
+
+                    if (hasEnvironmentVariables)
+                    {
+                        path = Interop.Crypto.GetX509RootStorePath();
+                        if (!Directory.Exists(path))
+                        {
+                            // X509 store ignores non-existing.
+                            path = string.Empty;
+                        }
+
+                        info = Interop.Crypto.GetX509RootStoreFile();
+                        if (!File.Exists(info))
```

Comment:
This means we're hitting the disk potentially multiple extra times per request?

Reply:
This whole function is a static lazy tuple.  So it's one extra File.Exists / Directory.Exists per process when the environment variables are in use.

Reply:
The GetX509RootStore\* functions are idempotent if the environment variables don't change.  So this seems "idempotent enough" to me.  

Even if this were in a check/lock/check/set it could still get sheared from X509Store if the environment variable is being written to during execution; so the (lack of) locking here seems good enough to me.

Reply:
Ok